### PR TITLE
Feat: EXPOSED-396 Supports fetchBatchedResults with sorting order 

### PR DESCRIPTION
**Pull request by roharon**
This PR includes `fetchBatchedResults` method which supports the `sortOrder` parameter.

Previously, the `fetchBatchedResults` method only supported the `batchSize` parameter, making it impossible to handle sorting order as it defaulted to ascending.
Adding support for the `sortOrder` parameter will be helpful.

```kotlin
val cities = DMLTestsData.Cities

cities.selectAll()
      .fetchBatchedResults(batchSize = 25, sortOrder = SortOrder.DESC)
```

### CODE REVIEW

**Comment by roharon** [comment]
I Failed on it.

But I guess `fetchBatchedResults` sql query don't need to preserve query as old query dsl. 
Because in query`SELECT Cities.`name` FROM Cities WHERE (Cities.`name` = 'andrey') AND (Cities.cityId > 0) ORDER BY Cities.cityId ASC LIMIT 50`, `(Cities.cityId > 0)` seems like is not necessary.


https://github.com/JetBrains/Exposed/blob/8ab66979392374f78a76ed36e2527e09cf8e3364/exposed-tests/src/test/kotlin/org/jetbrains/exposed/sql/tests/shared/dml/SelectTests.kt#L50

```sql
Iterable elements differ at index 6. Expected element <SELECT Cities.`name` FROM Cities WHERE (Cities.`name` = 'andrey') AND (Cities.cityId > 0) ORDER BY Cities.cityId ASC LIMIT 50>, actual element <SELECT Cities.`name` FROM Cities WHERE Cities.`name` = 'andrey' ORDER BY Cities.cityId ASC LIMIT 50>.
java.lang.AssertionError: Iterable elements differ at index 6. Expected element <SELECT Cities.`name` FROM Cities WHERE (Cities.`name` = 'andrey') AND (Cities.cityId > 0) ORDER BY Cities.cityId ASC LIMIT 50>, actual element <SELECT Cities.`name` FROM Cities WHERE Cities.`name` = 'andrey' ORDER BY Cities.cityId ASC LIMIT 50>.
```

**Review by bog-walk — CHANGES_REQUESTED** [review]
Hi @roharon Thanks for putting together a PR so quickly.

I've requested a small refactoring of the main logic.
The alternative to cover the failing test would be to attempt to duplicate the new logic into the old `FieldSet.selectBatched()`, but the commented suggestion would be preferable.

File: exposed-core/src/main/kotlin/org/jetbrains/exposed/sql/Query.kt [hunk 1]
```diff
@@ -260,16 +262,26 @@ open class Query(override var set: FieldSet, where: Op<Boolean>?) : AbstractQuer
...
+                                if (it == null) return@let whereOp
+
+                                return@let if (listOf(SortOrder.ASC, SortOrder.ASC_NULLS_FIRST, SortOrder.ASC_NULLS_LAST)
+                                        .contains(sortOrder)
+                                ) {
+                                    whereOp and (autoIncColumn greater it)
+                                } else {
+                                    whereOp and (autoIncColumn less it)
+                                }
+                            }
```
[the remark this answers is not in the document]
Comment (roharon) [reply exposed-core/src/main/kotlin/org/jetbrains/exposed/sql/Query.kt]: Thanks for kindly suggestion. It also looks good for me.

[the remark this answers is not in the document]
Comment (roharon) [reply exposed-core/src/main/kotlin/org/jetbrains/exposed/sql/Query.kt]: I applied it
[f748e91](https://github.com/JetBrains/Exposed/pull/2102/commits/f748e91153e16e55c9a9bc99dcf6ffd1452a242b)
